### PR TITLE
Defer `EditorInspector::update_tree` to the process stage to improve performance

### PR DESCRIPTION
**Pull request by Rindbee**
`EditorInspector::update_tree` is expensive, so defer the call to the process phase to prevent multiple calls in a single frame (when switching scene tabs).

Improve the experience when using `Ctrl + D` to copy nodes.

### CODE REVIEW

File: editor/editor_inspector.cpp [hunk 1]
```diff
@@ -3889,10 +3893,9 @@ void EditorInspector::_notification(int p_what) {
 			changing++;
 
 			if (update_tree_pending) {
-				update_tree();
+				// If there is a new update tree request during _update_tree, defer it to the next frame to ensure that the built tree is eventually correct.
 				update_tree_pending = false;
-				pending.clear();
```
Comment (KoBeWi) [inline editor/editor_inspector.cpp:3894]: Why did you remove `pending.clear()`? AFAIK it makes single properties refresh and if whole inspector is refreshed, no point in updating the properties further.

Comment (Rindbee) [reply editor/editor_inspector.cpp:3894]: Yep, thanks for pointing it out.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by TokageItLab** [comment]
This change seems to cause problems if the user selection in the inspector affects other editors, see #70588.
